src/annotator/integrations/pdf.js

```diff
@@ -210,29 +227,33 @@ export class PDFIntegration {
 
     try {
       const hasText = await documentHasText();
-      this._toggleNoSelectableTextWarning(!hasText);
+      this._updateBannerState({ noTextWarning: !hasText });
     } catch (err) {
       /* istanbul ignore next */
       console.warn('Unable to check for text in PDF:', err);
     }
   }
 
   /**
-   * Set whether the warning about a PDF's suitability for use with Hypothesis
-   * is shown.
+   * Update banners shown above the PDF viewer.
    *
-   * @param {boolean} showWarning
+   * @param {Partial<typeof PDFIntegration.prototype._bannerState>} state
```

Comment:
This typing leaves me scratching my head a little bit...

Reply:
This is defining the type of an internal function's parameters in terms of a field on the class. I wouldn't do this for a public method, but this is private.